### PR TITLE
feat(server-openapi): provide public methods #generateOpenApiAsYaml/Json

### PR DESCRIPTION
**Pull request by christopher-cudennec**
it can be used to generate the documentation without a running application

### CODE REVIEW

File: sda-commons-server-openapi/src/main/java/org/sdase/commons/server/openapi/OpenApiBundle.java [hunk 1]
```diff
@@ -133,6 +124,50 @@ public void run(Configuration configuration, Environment environment) {
...
+    return result;
+  }
+
+  /**
+   * This method can be used to generate the OpenAPI specification as JSON or YAML.
+   *
+   * @param type the type of the specification, either 'json' or 'yaml'
+   * @return the OpenAPI specification as JSON or YAML
+   */
+  public String generateOpenApi(String type) throws Exception {
```
Comment (JoergSiebahn) [inline sda-commons-server-openapi/src/main/java/org/sdase/commons/server/openapi/OpenApiBundle.java:149]: I would prefer two dedicated methods `generateOpenApiAsYaml()` and `generateOpenApiAsJson()` (or only yaml) so I don't have to mention, that the parameter should be an enum.

File: docs/server-open-api.md [hunk 2]
```diff
@@ -58,40 +58,32 @@ To automatically generate the OpenAPI spec and ensure that it is committed to ve
...
   @Test
-  void shouldHaveSameOpenApiInRepository() throws IOException {
-    // receive the openapi.yaml from your service
-    String expected =
-        DW.client()
-            .target(String.format("http://localhost:%s", DW.getLocalPort()))
-            .path("openapi.yaml")
-            .request()
-            .get(String.class);
+  void shouldGenerateDocumentationWithoutRestCall() throws Exception {
```
Comment (JoergSiebahn) [inline docs/server-open-api.md:72]: Maybe we should add this example diff to the release notes (manually after release).

File: sda-commons-server-openapi/src/main/java/org/sdase/commons/server/openapi/OpenApiBundle.java (hunk 1 above)
Comment (christopher-cudennec) [reply sda-commons-server-openapi/src/main/java/org/sdase/commons/server/openapi/OpenApiBundle.java]: Sure, no problem. It's less guessing and effort for consumers if they conveniently choose between the two. 👍

File: docs/server-open-api.md (hunk 2 above)
Comment (christopher-cudennec) [reply docs/server-open-api.md]: I can see what I + Markdown can do 😄

Comment (JoergSiebahn) [reply docs/server-open-api.md]: Markdown and GitHub are best friends:

```diff
 class OpenApiDocumentationTest {
-    // start your application
-    @RegisterExtension
-    public static final DropwizardAppExtension<Configuration> DW =
-            new DropwizardAppExtension<>(YourApplication.class, resourceFilePath("test-config.yaml"));
 
     @Test
-    void shouldHaveSameOpenApiInRepository() throws IOException {
-        // receive the openapi.yaml from your service
-        String expected =
-                DW.client()
-                        .target(String.format("http://localhost:%s", DW.getLocalPort()))
-                        .path("openapi.yaml")
-                        .request()
-                        .get(String.class);
+    void shouldHaveSameOpenApiInRepository() throws Exception {
+        // receive the openapi.yaml from the OpenApiBundle
+        var bundle =
+                OpenApiBundle.builder()
+                        .addResourcePackage(YourApplication.class.getPackageName())
+                        .build();
+        var openapi = bundle.generateOpenApiAsYaml();
+        assertThat(openapi).isNotNull();
 
         // specify where you want your file to be stored
         Path filePath = Paths.get("openapi.yaml");
 
         // check and update the file
         GoldenFileAssertions.assertThat(filePath)
-                .hasContentAndUpdateGolden(normalizeOpenApiYaml(expected));
+                .hasYamlContentAndUpdateGolden(normalizeOpenApiYaml(openapi));
     }
 }
```

Source:
<pre><code>```diff
 class OpenApiDocumentationTest {
-    // start your application
-    @RegisterExtension
-    public static final DropwizardAppExtension<Configuration> DW =
-            new DropwizardAppExtension<>(YourApplication.class, resourceFilePath("test-config.yaml"));
 
     @Test
-    void shouldHaveSameOpenApiInRepository() throws IOException {
-        // receive the openapi.yaml from your service
-        String expected =
-                DW.client()
-                        .target(String.format("http://localhost:%s", DW.getLocalPort()))
-                        .path("openapi.yaml")
-                        .request()
-                        .get(String.class);
+    void shouldHaveSameOpenApiInRepository() throws Exception {
+        // receive the openapi.yaml from the OpenApiBundle
+        var bundle =
+                OpenApiBundle.builder()
+                        .addResourcePackage(YourApplication.class.getPackageName())
+                        .build();
+        var openapi = bundle.generateOpenApiAsYaml();
+        assertThat(openapi).isNotNull();
 
         // specify where you want your file to be stored
         Path filePath = Paths.get("openapi.yaml");
 
         // check and update the file
         GoldenFileAssertions.assertThat(filePath)
-                .hasContentAndUpdateGolden(normalizeOpenApiYaml(expected));
+                .hasYamlContentAndUpdateGolden(normalizeOpenApiYaml(openapi));
     }
 }
```
</code></pre>

**Review by JoergSiebahn — APPROVED** [review]

**Pull request merged**